zeebe/qa/util/src/main/java/io/camunda/zeebe/qa/util/junit/ZeebeIntegrationExtension.java

```diff
@@ -250,97 +250,102 @@ private Path createManagedDirectory(final Store store, final String prefix) {
   }
 
   private ClusterResource asClusterResource(final Object testInstance, final Field field) {
-    final TestCluster value;
+    ReflectUtil.makeAccessible(field, testInstance);
 
-    try {
-      value = (TestCluster) ReflectUtil.makeAccessible(field, testInstance).get(testInstance);
-    } catch (final IllegalAccessException e) {
-      throw new UnsupportedOperationException(e);
-    }
-
-    return new ClusterResource(value, field.getAnnotation(TestZeebe.class));
+    return new ClusterResource(testInstance, field, field.getAnnotation(TestZeebe.class));
   }
 
   private ApplicationResource asNodeResource(final Object testInstance, final Field field) {
-    final TestApplication<?> value;
+    ReflectUtil.makeAccessible(field, testInstance);
 
-    try {
-      value =
-          (TestApplication<?>) ReflectUtil.makeAccessible(field, testInstance).get(testInstance);
-    } catch (final IllegalAccessException e) {
-      throw new UnsupportedOperationException(e);
-    }
-
-    return new ApplicationResource(value, field.getAnnotation(TestZeebe.class));
+    return new ApplicationResource(testInstance, field, field.getAnnotation(TestZeebe.class));
   }
 
   private Store store(final ExtensionContext extensionContext) {
     return extensionContext.getStore(Namespace.create(ZeebeIntegrationExtension.class));
   }
 
-  private record ClusterResource(TestCluster cluster, TestZeebe annotation)
+  private record ClusterResource(Object testInstance, Field field, TestZeebe annotation)
       implements TestZeebeResource, CloseableResource {
 
+    public TestCluster cluster() {
+      try {
+        return (TestCluster) field.get(testInstance);
```

Comment:
:thought_balloon: Can we/should we memoize this value? It doesn't hurt too much anyway since this isn't exactly in the hot path, but just wondering.

Reply:
Considering the performance are not an issue, I assumed getting the field from the class each  time can avoid potential bugs (such as memoizing null)